app/src/main/java/kth/jjve/xfran/WorkoutsActivity.java

```diff
@@ -14,97 +14,56 @@
 
-public class WorkoutsTabActivity extends AppCompatActivity implements NavigationView.OnNavigationItemSelectedListener, WorkoutsRecyclerAdapter.ListItemClickListener {
+public class WorkoutsActivity extends BaseActivity implements WorkoutsRecyclerAdapter.ListItemClickListener {
 
     private final String LOG_TAG = getClass().getSimpleName();
 
     /*_________ VIEW MODEL _________*/
-    private WorkoutsViewModel mWorkoutsViewModel;
+    private WorkoutsVM mWorkoutsViewModel;
     private WorkoutsRecyclerAdapter mAdapter;
 
-    /*_________ VIEW _________*/
-    private RecyclerView mRecyclerView;
-    private DrawerLayout drawerLayout;
-    private NavigationView navigationView;
-    private Toolbar toolbar;
-
     /*_________ INTENT _________*/
     public static String WORKOUT_ID="Workout ID";
     public static String WORKOUT_OBJ="Workout Obj";
 
     @Override
     protected void onCreate(Bundle savedInstanceState) {
         super.onCreate(savedInstanceState);
-        setContentView(R.layout.activity_workouts_tab);
+
+        FrameLayout contentFrameLayout = findViewById(R.id.content_frame);
+        getLayoutInflater().inflate(R.layout.act_workouts, contentFrameLayout);
+        navigationView.setCheckedItem(R.id.nav_workouts);
 
         /*------ HOOKS ------*/
-        drawerLayout = findViewById(R.id.drawerlayout_trainingdiary);
-        mRecyclerView = findViewById(R.id.rv_trainingdiary);
-        navigationView = findViewById(R.id.nav_view_diary);
-        toolbar = findViewById(R.id.diary_toolbar);
+        /*_________ VIEW _________*/
+        RecyclerView mRecyclerView = findViewById(R.id.rv_trainingdiary);
 
         /*----- VIEW MODEL -----*/
-        mWorkoutsViewModel = ViewModelProviders.of(this).get(WorkoutsViewModel.class);
+        mWorkoutsViewModel = ViewModelProviders.of(this).get(WorkoutsVM.class);
         mWorkoutsViewModel.init();
         mWorkoutsViewModel.getWorkouts().observe(this, workouts -> mAdapter.notifyDataSetChanged());
 
         /*------ INIT ------*/
-        setSupportActionBar(toolbar);
-        initNavMenu();
-        initRecyclerView();
-
-    }
-
-    @Override
-    protected void onResume(){
-        super.onResume();
-        navigationView.setCheckedItem(R.id.nav_workouts);
-    }
-
-    @Override
-    public void onBackPressed() {
-        if(drawerLayout.isDrawerOpen(GravityCompat.START)){
-            drawerLayout.closeDrawer(GravityCompat.START);
-        } else {
-            super.onBackPressed();
-        }
-    }
-
-    @Override
-    public boolean onNavigationItemSelected(@NonNull MenuItem menuItem) {
-        int id = menuItem.getItemId();
-        if (id == R.id.nav_home){
-            Intent intent = new Intent(WorkoutsTabActivity.this, HomeScreenActivity.class);
-            startActivity(intent);
-            finish();
-        }
-        drawerLayout.closeDrawer(GravityCompat.START);
-        return true;
-    }
-
-    private void initRecyclerView(){
```

Comment:
kinda confused about what's happening here, wouldn't we want to keep the method initRecyclerView an call it in onCreate?

Reply:
It has to do with readability (same issue I mentioned before). 

We might wanna discuss it in the meeting tonight and make a conclusion on what to do. In my eyes, creating seperate methods does help to clean up the onCreate but it also decreases readability as you have to scroll back and forth to maybe understand what is happening. As this is only 4 lines I decided for now that we can just put it in the onCreate (since the onCreate is also only 20 lines)

Reply:
A short bit of research showed me that there is no clear convention --> so we can make our own 😄